Application/Education/Certificate/Generator/Service/DataCertificate/IDataFELS.php

```diff
@@ -0,0 +1,278 @@
+                        $Data->createCertificateLevel($tblCertificate, $tblLevel);
+                    }
+                    if (($tblLevel = Division::useService()->getLevelBy($Data->getTblSchoolTypeGym(), '7'))) {
+                        $Data->createCertificateLevel($tblCertificate, $tblLevel);
+                    }
+                    if (($tblLevel = Division::useService()->getLevelBy($Data->getTblSchoolTypeGym(), '8'))) {
+                        $Data->createCertificateLevel($tblCertificate, $tblLevel);
+                    }
+                    if (($tblLevel = Division::useService()->getLevelBy($Data->getTblSchoolTypeGym(), '9'))) {
+                        $Data->createCertificateLevel($tblCertificate, $tblLevel);
+                    }
+                    if (($tblLevel = Division::useService()->getLevelBy($Data->getTblSchoolTypeGym(), '10'))) {
+                        $Data->createCertificateLevel($tblCertificate, $tblLevel);
+                    }
+                }
+            }
+            // Begrenzung des Bemerkungsfeld
+            $FieldName = 'Remark';
+            if (!$Data->getCertificateFieldByCertificateAndField($tblCertificate, $FieldName)){
+                $Data->createCertificateField($tblCertificate, $FieldName, 600);
+            }
+        }
+        if ($tblCertificate && !$Data->getCertificateGradeAll($tblCertificate)) {
+            $Data->setCertificateGradeAllStandard($tblCertificate);
+        }
+        if ($tblCertificate && !$Data->getCertificateSubjectAll($tblCertificate)) {
+            $Data->setCertificateSubject($tblCertificate, 'DE', 1, 1);
+            $Data->setCertificateSubject($tblCertificate, 'EN', 1, 2);
+            // 1,3 freilassen für Fremdsprache
+            $Data->setCertificateSubject($tblCertificate, 'KU', 1, 4);
+            $Data->setCertificateSubject($tblCertificate, 'MU', 1, 5);
+            $Data->setCertificateSubject($tblCertificate, 'GE', 1, 6);
+            $Data->setCertificateSubject($tblCertificate, 'GRW', 1, 7);
+            $Data->setCertificateSubject($tblCertificate, 'GEO', 1, 8);
+
+            $Data->setCertificateSubject($tblCertificate, 'MA', 2, 1);
+            $Data->setCertificateSubject($tblCertificate, 'BIO', 2, 2);
+            $Data->setCertificateSubject($tblCertificate, 'CH', 2, 3);
+            $Data->setCertificateSubject($tblCertificate, 'PH', 2, 4);
+            $Data->setCertificateSubject($tblCertificate, 'SPO', 2, 5);
+            $Data->setCertificateSubject($tblCertificate, 'RE/e', 2, 6);
+            $Data->setCertificateSubject($tblCertificate, 'TC', 2, 7);
+            $Data->setCertificateSubject($tblCertificate, 'INF', 2, 8);
+        }
+    }
+    /**
+     * @param Data        $Data
+     * @param TblConsumer $tblConsumerCertificate
+     */
+    private static function setGymJ(Data $Data, TblConsumer $tblConsumerCertificate)
```

Comment:
Codacy found an issue: [The method setGymJ() has a Cyclomatic Complexity of 16. The configured cyclomatic complexity threshold is 10.](https://app.codacy.com/manual/gerdchristian-kunze/SPHERE-Framework/pullRequest?prid=5757982)